velox/duckdb/conversion/DuckConversion.h

```diff
@@ -87,6 +87,41 @@ struct DuckStringConversion {
   }
 };
 
+struct DuckShortDecimalConversion {
+  typedef int64_t DUCK_TYPE;
+  typedef ShortDecimal VELOX_TYPE;
+
+  static int64_t toDuck(
+      const ShortDecimal& input,
+      ::duckdb::Vector& /* unused */) {
+    return input.unscaledValue();
+  }
+
+  template <typename T>
+  static ShortDecimal toVelox(const T input) {
+    return ShortDecimal(input);
+  }
+};
+
+struct DuckLongDecimalConversion {
+  typedef ::duckdb::hugeint_t DUCK_TYPE;
+  typedef LongDecimal VELOX_TYPE;
+
+  static ::duckdb::hugeint_t toDuck(
+      const LongDecimal& input,
+      ::duckdb::Vector& /* unused */) {
+    ::duckdb::hugeint_t duckValue;
+    duckValue.upper = (input.unscaledValue() >> 64);
+    duckValue.lower = input.unscaledValue();
+    return duckValue;
+  }
+
+  template <typename T>
+  static LongDecimal toVelox(const T input) {
```

Comment:
Does this have to be a template? 
Input here can only be `const DUCK_TYPE &` correct?

Reply:
Based on other implementations, this should be 
`static double toVelox(const ::duckdb::hugeint_t& input) {`